whylog/tests/utils.py

```diff
@@ -8,3 +11,33 @@ def get_path_to_config_files(cls, prefix_path):
         rules_path = os.path.join(prefix_path, 'rules.yaml')
         log_type_path = os.path.join(prefix_path, 'log_types.yaml')
         return [parsers_path, rules_path, log_type_path]
+
+
+class TestRemovingSettings(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        #This change was caused by fails os.names() for new directories.
+        #Renaming was necessary to avoid conflicts between test config directories
+        #and orginal independent non tests config directories
+        SettingsFactorySelector.WHYLOG_DIR = '.whylog_test'
+        cls.remove_settings_dirs()
+
+    @classmethod
+    def tearDownClass(cls):
+        #Removed test config directories if test failed
+        cls.remove_settings_dirs()
+
+    @classmethod
+    def remove_one(clas, path):
+        config_dir = os.path.join(path, SettingsFactorySelector.WHYLOG_DIR)
+        if os.path.isdir(config_dir):
+            shutil.rmtree(config_dir)
+
+    @classmethod
+    def remove_settings_dirs(cls):
+        path = os.getcwd()
+        cls.remove_one(path)
+        for i in range(2):
+            path, _ = os.path.split(path)
+            cls.remove_one(path)
```

Comment:
How do you guarantee, that this will not remove user home directory? If I understand this code correctly, if starting path is for example: `~/foo/bar`, you will attempt to remove at first `~/foo/bar` (line 39), then `/foo` (line 42, first time) and finally `~` (line 42 second time)!